File: java/app-encryption/src/it/java/com/godaddy/asherah/utils/PayloadGenerator.java
package com.godaddy.asherah.utils;

import java.nio.charset.StandardCharsets;

import org.apache.commons.text.RandomStringGenerator;
import org.json.JSONObject;

public final class PayloadGenerator {
  private static final RandomStringGenerator RANDOM_STRING_GENERATOR = new RandomStringGenerator.Builder()
    .withinRange('0', 'z')
    .filteredBy(Character::isLetterOrDigit)
    .build();
  private static final int DEFAULT_BYTE_SIZE = 20;

  private PayloadGenerator() {
  }

  public static byte[] createDefaultRandomBytePayload() {
    return createRandomBytePayload(DEFAULT_BYTE_SIZE);
  }

  public static byte[] createRandomBytePayload(final int size) {
    return RANDOM_STRING_GENERATOR.generate(size).getBytes(StandardCharsets.UTF_8);
  }

  public static JSONObject createDefaultRandomJsonPayload() {
    return createRandomJsonPayload(DEFAULT_BYTE_SIZE);
  }

  public static JSONObject createRandomJsonPayload(final int size) {
    // This will end up having an extra 10 bytes from json overhead + key, meh
    JSONObject json = new JSONObject();
    json.put("key", RANDOM_STRING_GENERATOR.generate(size));
    return json;
  }
}

File: csharp/AppEncryption/AppEncryption.IntegrationTests/Utils/PayloadGenerator.cs
using System;
using System.Text;
using Newtonsoft.Json.Linq;

namespace GoDaddy.Asherah.AppEncryption.IntegrationTests.Utils
{
    public static class PayloadGenerator
    {
        private const int DefaultByteSize = 20;
        private const string PermittedCharacters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";

        private static readonly Random Random = new Random();

        public static byte[] CreateDefaultRandomBytePayload()
        {
            return CreateRandomBytePayload(DefaultByteSize);
        }

        public static byte[] CreateRandomBytePayload(int size)
        {
            return Encoding.UTF8.GetBytes(RandomStringGenerator(size, Random));
        }

        public static JObject CreateDefaultRandomJsonPayload()
        {
            return CreateRandomJsonPayload(DefaultByteSize);
        }

        public static JObject CreateRandomJsonPayload(int size)
        {
            // This will end up having an extra 10 bytes from json overhead + key, meh
            JObject json = new JObject();
            json.Add("key", RandomStringGenerator(size, Random));
            return json;
        }

        private static string RandomStringGenerator(int length, Random random)
        {
            StringBuilder result = new StringBuilder(length);
            for (int i = 0; i < length; i++)
            {
                result.Append(PermittedCharacters[random.Next(PermittedCharacters.Length)]);
            }

            return result.ToString();
        }
    }
}
